src/state_point.F90

```diff
@@ -126,151 +147,197 @@ subroutine write_state_point()
-                 group="tallies/tally" // trim(to_str(i)) // "/moments")
-            k = k + 1
-          case (SCORE_SCATTER_PN, SCORE_NU_SCATTER_PN)
-            do n_order = 0, t % moment_order(k)
-              moment_name = 'P' // trim(to_str(n_order))
+      if (n_tallies > 0) then
+
+        ! Print list of tally IDs
+        allocate(id_array(n_tallies))
+        allocate(key_array(n_tallies))
+
+        ! Write all tally information except results
+        do i = 1, n_tallies
+          tally => tallies(i)
+          key_array(i) = tally % id
+          id_array(i) = i
+        end do
+
+        call sp % write_data(id_array, "ids", &
+             group="tallies", length=n_tallies)
+        call sp % write_data(key_array, "keys", &
+             group="tallies", length=n_tallies)
+
+        deallocate(key_array)
+
+        ! Write all tally information except results
+        TALLY_METADATA: do i = 1, n_tallies
+
+          ! Get pointer to tally
+          tally => tallies(i)
+
+          call sp % write_data(len(tally % label), "label_size", &
+               group="tallies/tally " // trim(to_str(tally % id)))
+          if (len(tally % label) > 0) then
+            call sp % write_data(tally % label, "label", &
+                 group="tallies/tally " // trim(to_str(tally % id)))
+          endif
+
+          call sp % write_data(tally % estimator, "estimator", &
+               group="tallies/tally " // trim(to_str(tally % id)))
+          call sp % write_data(tally % n_realizations, "n_realizations", &
+               group="tallies/tally " // trim(to_str(tally % id)))
+          call sp % write_data(tally % n_filters, "n_filters", &
+               group="tallies/tally " // trim(to_str(tally % id)))
+
+          ! Write filter information
+          FILTER_LOOP: do j = 1, tally % n_filters
+
+            call sp % write_data(tally % filters(j) % type, "type", &
+                 group="tallies/tally " // trim(to_str(tally % id)) // &
+                 "/filter " // to_str(j))
+            call sp % write_data(tally % filters(j) % n_bins, "n_bins", &
+                 group="tallies/tally " // trim(to_str(tally % id)) // &
+                 "/filter " // to_str(j))
+            if (tally % filters(j) % type == FILTER_ENERGYIN .or. &
+                tally % filters(j) % type == FILTER_ENERGYOUT) then
+              call sp % write_data(tally % filters(j) % real_bins, "bins", &
+                   group="tallies/tally " // trim(to_str(tally % id)) // &
+                   "/filter " // to_str(j), &
+                   length=size(tally % filters(j) % real_bins))
+            else
+              call sp % write_data(tally % filters(j) % int_bins, "bins", &
+                   group="tallies/tally " // trim(to_str(tally % id)) // &
+                   "/filter " // to_str(j), &
+                   length=size(tally % filters(j) % int_bins))
+            end if
```

Comment:
For a tally with a mesh filter, the `int_bins` attribute currently stores the index of the mesh in the meshes array. This is set on line 2145 of `input_xml.F90`. According to the documentation, the bins attribute should be the `id` of the mesh used as a filter, not the index of the mesh in the meshes array. A simple fix would be to add the following code after line 271:

elseif (tally % filters(j) % type == FILTER_MESH) then
    mesh => meshes(tally % filters(j) % int_bins(1))
    call sp % write_data(mesh % id, "bins", &
        group="tallies/tally " // trim(to_str(tally % id)) // &
        "/filter " // to_str(j))

Another option could be to change the value of `int_bins` set in `input_xml.F90` to the mesh `id`. This might be considered to be more in line with the documentation, but would probably require other changes to be incorporated in places where the `int_bins` attribute is currently used to index into the meshes array.